test/functional/fixtures/native-dialogs-handling/test.js

```diff
@@ -0,0 +1,156 @@
+var expect                                    = require('chai').expect;
+var errorInEachBrowserContains                = require('../../assertion-helper.js').errorInEachBrowserContains;
+var getNativeDialogNotHandledErrorText        = require('./errors.js').getNativeDialogNotHandledErrorText;
+var getUncaughtErrorInNativeDialogHandlerText = require('./errors.js').getUncaughtErrorInNativeDialogHandlerText;
+
+
+describe('Native dialogs handling', function () {
+    describe('Errors during dialogs handling', function () {
+        it('Should fail if an unexpected confirm dialog appears after an action', function () {
+            return runTests('./testcafe-fixtures/native-dialogs-test.js', 'Without handler',
+                { shouldFail: true, skipJsErrors: true })
+                .catch(function (errs) {
+                    errorInEachBrowserContains(errs, getNativeDialogNotHandledErrorText('confirm'), 0);
+                    errorInEachBrowserContains(errs, '> 14 |    await t.click(\'#buttonConfirm\'); ', 0);
+                });
+        });
+
+        it('Should pass if the expected confirm dialog appears after an action', function () {
+            return runTests('./testcafe-fixtures/native-dialogs-test.js', 'Expected confirm after an action');
+        });
+
+        it('Should pass if the expected confirm dialog appears after an action (with dependencies)', function () {
+            return runTests('./testcafe-fixtures/native-dialogs-test.js', 'Expected confirm after an action (with dependencies)');
+        });
+
+        it('Should pass if the expected confirm dialog appears after an action (client function)', function () {
+            return runTests('./testcafe-fixtures/native-dialogs-test.js', 'Expected confirm after an action (client function)');
+        });
+
+        it('Should pass if different dialogs appear after an actions', function () {
```

Comment:
> an actions

just `actions`?
That's because I used to blow your mind with those articles, I know, sorry =)